### ISSUE TITLE
Add troubleshooting section for the article "Secure the Kubeflow authentication with HTTPS" on IBM Cloud

### ISSUE THREAD

**Issue by shawnzhu** [kubeflow/website#2353]
I've got several cases that the guide [Secure the Kubeflow authentication with HTTPS](https://www.kubeflow.org/docs/ibm/deploy/authentication/) is really hard to follow and caused trouble that's hard to debug.

@animeshsingh suggested adding a troubleshooting section and it sounds good idea.

**Comment by soumochak83** [comment]
Hi! Sharing my experience to secure Kubeflow 1.1 on IBM Cloud IKS using AppID.

I followed the documentation as available on Kubeflow page - https://www.kubeflow.org/docs/ibm/deploy/install-kubeflow/#multi-user-auth-enabled ...but after configuring every step as expected, I couldn't connect to Kubeflow using AppID. Whenever I tried hitting the Kubeflow URL as configured in the steps, the browser returned an error..."Could not connect".
I then ran a CURL on command line as: curl -v https://<kubeflow url> and found the below message:
About to connect() to <kubeflow url>.us-south.containers.appdomain.cloud port 443 (#0)
*   Trying 169.xx.xx.xx...
* Connection refused

This raised suspicion about the ingress gateway pod in istio-system namespace if its picking up the correct config. I ran the command: kubectl logs -f <istio-ingressgateway pod name> -n istio-system and it retuned the below error:

[2020-11-11 15:21:55.364][20][warning][config] [external/envoy/source/common/config/grpc_mux_subscription_impl.cc:81] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener(s) 0.0.0.0_443: Invalid path: /etc/istio/ingressgateway-certs/tls.crt
[2020-11-11 15:22:41.394][20][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13, 
[2020-11-11 15:22:41.541][20][warning][config] [external/envoy/source/common/config/grpc_mux_subscription_impl.cc:81] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener(s) 0.0.0.0_443: Invalid path: /etc/istio/ingressgateway-certs/tls.crt

Now, the issue was very clear from error above - the certificates were missing in the istio-ingressgateway pod. I tried to check if the secret was deployed using the command: kubectl describe secret/istio-ingressgateway-certs -n istio-system and the result was: Error from server (NotFound): secrets "istio-ingressgateway-certs" not found
Hence, it was very evident I missed a step to create the istio-ingressgateway-certs certificate from Step 6 of the documentation - https://www.kubeflow.org/docs/ibm/deploy/authentication/#exposing-the-kubeflow-dashboard-with-dns-and-tls-termination
After creating the certificate, rescheduled the istio-ingressgateway using the command kubectl delete pod/istio-ingressgateway-xxx -n istio-system --force --grace-period=0 and then tried to trigger a CURL to the Kubeflow URL. It connected and issue was fixed!

Lessons learnt: The steps are quite exhaustive and its highly likely to miss a point since they are not highlighted sufficiently (at least the important ones). So, once you deploy the entire config, review them step by step and ensure the components are in place and working as expected. It may take little while but its the only way for a perfect deployment.

Wishlist: It would be nice to add a "Logout" button in the Kubeflow UI. That would add flexibility to end user session.
